cli/flox/src/commands/environment.rs

```diff
@@ -1059,22 +1102,41 @@ impl Push {
         let path_environment =
             path_environment::PathEnvironment::open(path_pointer, dot_flox_path, &flox.temp_dir)?;
 
-        let env =
-            ManagedEnvironment::push_new(flox, path_environment, owner.parse().unwrap(), force)
-                .map_err(Self::convert_error)?;
+        let pointer = ManagedPointer::new(owner.clone(), path_environment.name(), &flox.floxhub);
+
+        let env = ManagedEnvironment::push_new(flox, path_environment, owner, force)
+            .map_err(|err| Self::convert_error(err, pointer, true))?;
 
         Ok(env)
     }
 
-    fn convert_error(err: ManagedEnvironmentError) -> anyhow::Error {
-        if let ManagedEnvironmentError::OpenFloxmeta(FloxmetaV2Error::LoggedOut) = err {
-            anyhow!(indoc! {"
-                Could not push environment: not logged in to floxhub.
+    fn convert_error(
+        err: ManagedEnvironmentError,
+        pointer: ManagedPointer,
+        create_remote: bool,
+    ) -> anyhow::Error {
+        let owner = &pointer.owner;
+        let name = &pointer.name;
 
-                Please login to floxhub with `flox auth login`
-                "})
+        let message = match err {
+            ManagedEnvironmentError::AccessDenied => formatdoc! {"
+                ❌  You do not have permission to write to {owner}/{name}
+            "}.into(),
+            ManagedEnvironmentError::Diverged if create_remote => formatdoc! {"
+                ❌  An environment named {owner}/{name} already exists!
```

Comment:
question nonblocking: (this is contradictory to the ticket so no need to change) do we need to mention the possibility of the environment having diverged? It's somewhat implied by the `flox delete && flox pull` but not super obvious.  @ghudgins